app/grandchallenge/cases/tasks.py

```diff
@@ -159,20 +165,43 @@ def build_images(  # noqa:C901
         _delete_session_files(upload_session=upload_session)
         upload_session.update_status(
             status=RawImageUploadSession.FAILURE,
-            error_message=str(e),
+            error_message=(
+                "One or more of the inputs failed validation."
+                if linked_object
+                else str(e)
+            ),
+            detailed_error_message=(
+                {ci.title: str(e)} if linked_object else None
+            ),
             linked_object=linked_object,
         )
     except (SoftTimeLimitExceeded, TimeLimitExceeded):
         upload_session.update_status(
             status=RawImageUploadSession.FAILURE,
-            error_message="Time limit exceeded",
+            error_message=(
+                "One or more of the inputs failed validation."
+                if linked_object
+                else "Time limit exceeded"
+            ),
+            detailed_error_message=(
+                {ci.title: "Time limit exceeded"} if linked_object else None
+            ),
             linked_object=linked_object,
         )
     except Exception:
         _delete_session_files(upload_session=upload_session)
         upload_session.update_status(
             status=RawImageUploadSession.FAILURE,
-            error_message="An unexpected error occurred",
+            error_message=(
+                "One or more of the inputs failed validation."
+                if linked_object
+                else "An unexpected error occurred"
+            ),
+            detailed_error_message=(
+                {ci.title: "An unexpected error occurred"}
+                if linked_object
+                else None
+            ),
```

Comment:
Could these repeated `if`/`else` statements be handled in `upload_session.update_status` instead. They only seem to depend on `linked_object` which `upload_session.update_status` has access to.